core-bundle/src/Resources/contao/classes/FrontendTemplate.php

```diff
@@ -356,24 +357,48 @@ private function setCacheHeaders(Response $response)
 			return $response->setPrivate();
 		}
 
-		// Do not cache the response if a user is logged in or the page is protected
-		// TODO: Add support for proxies so they can vary on member context
-		if (FE_USER_LOGGED_IN === true || BE_USER_LOGGED_IN === true || $objPage->protected || $this->hasAuthenticatedBackendUser())
+		// Do not cache response if any cookies were received or sent and make sure the response is private.
+		if (0 !== \count($response->headers->getCookies()) || 0 !== \count(System::getContainer()->get('request_stack')->getCurrentRequest()->headers->getCookies()))
```

Comment:
```suggestion
		if (0 !== \count($response->headers->getCookies()) || 0 !== System::getContainer()->get('request_stack')->getCurrentRequest()->cookies->count())
```

Reply:
Is `getCurrentRequest()` the right one here?

Reply:
Imho yes.